bases/rsptx/interactives/runestone/hparsons/js/hparsons.js

```diff
@@ -152,13 +160,64 @@ export default class HParsons extends RunestoneBase {
             that.hparsonsInput.resetInput();
             that.setLocalStorage();
             that.feedbackController.reset();
+            that.renderMathInBlocks();
         };
         $(resetBtn).attr("type", "button");
 
         $(this.outerDiv).prepend(ctrlDiv);
         this.controlDiv = ctrlDiv;
     }
 
+    // Decodes escaped HTML entities (like &lt;) into raw characters
+    decodeHTMLEntities(str) {
+        const textarea = document.createElement("textarea");
+        textarea.innerHTML = str;
+        return textarea.value;
+    }
+
+    renderMathInBlocks() {
+        if (this.language !== "math") return;
+        setTimeout(() => {
+            const blocks = document.querySelectorAll(`#${this.divid}-container .parsons-block`);
+            blocks.forEach(block => {
+                block.innerHTML = this.decodeHTMLEntities(block.innerHTML);
+            });
+
+            if (window.MathJax && MathJax.typesetPromise) {
```

Comment:
Thanks for this addition!

You should have a look at parsons.js and others for how we use the this.queueMathJax(specific item).

MathJax can create a big set of timing problems and this allows us to take control.  Also trying to re-typeset the whole page can be a performance issue.